finality-provider/service/chain_poller.go

```diff
@@ -74,7 +74,8 @@ func NewChainPoller(
 
 func (cp *ChainPoller) Start(startHeight uint64) error {
 	if cp.isStarted.Swap(true) {
-		return fmt.Errorf("the poller is already started")
+		cp.logger.Info("the poller is already started")
+		return nil
```

Comment:
Why change to return `nil`? If so, it will hide the fact that the poller is called twice, which we do not expect.